### PR TITLE
#159361808 Ratings and reviews

### PR DESCRIPTION
**Pull request by otseobande**
[//]: # (PR Title should follow this format "#<story-id> story description")

#### What does this PR do?
Implements ratings and review feature

#### Description of Task to be completed?
- add feature to allow users who has bought a product to give reviews
- add average reviews as stars to product card
- add a link to the order page for review
- paginate reviews

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots or gifs (if appropriate):

screen shot 2018-08-23 at 2 24 37 am

#### Questions:
N/A

### CODE REVIEW

**Review by iverenshaguy — COMMENTED** [review]
Nice work! It is really good that a vendor cannot review his own product. I'm also impressed that you took advantage of the existing list view to paginate the review. This shows that you paid attention to details. Keep it up!

I have a question though. Is it normal for a user to be able to add more than 1 review to a product. Shouldn't a new review be a modification of the former review? Or was it implemented that way on purpose?

File: imports/plugins/custom/gdr-landing-page/client/components/ProductsCategories.js
Comment (iverenshaguy) [inline imports/plugins/custom/gdr-landing-page/client/components/ProductsCategories.js]: I love your show of initiative in fixing this typo. Way to go!

**Comment by otseobande** [comment]
It was implemented that way on purpose. Initially, I implemented it in a way where a user can only review once and won't be able to review again. I got feedback that users are normally able to review more than once so I left it that way. Well, I'm not sure anymore though.

**Review by iverenshaguy — DISMISSED** [review]

File: imports/plugins/custom/gdr-landing-page/client/components/ProductsCategories.js
Comment (iverenshaguy) [reply imports/plugins/custom/gdr-landing-page/client/components/ProductsCategories.js]: Amazing work.

**Comment by iverenshaguy** [comment]
Oh! That's really interesting though. I think this is not so secure and can make ratings gimmicked. For instance, my friend bought my product once but can give it a 5 star 50 times, thereby giving me a higher rating. I think what the other person meant by more than 1 rating is commenting. Google play, for example, allows many comments UNDER your rating and review in case the app owner wants to communicate with you. Your rating and review, however, remains just one and can be modified anytime. I think we should look into understanding it more before merging this PR.

**Comment by iverenshaguy** [comment]
I'm sorry I can't approve this now. I'll change my approval to a request for a change. Let's talk it out as a team first.

**Review by ozimos — APPROVED** [review]
Outstanding work. Keep it coming

**Comment by otseobande** [comment]
Done

**Comment by otseobande** [comment]
Thanks Tovieye

File: imports/plugins/custom/product-variant/client/components/customProductGridItem.js
Comment (rovilay) [inline imports/plugins/custom/product-variant/client/components/customProductGridItem.js:86]: Good job on Code Refactoring. Keep it up.

File: imports/plugins/custom/reviews/client/components/ReviewsPagination.js
Comment (rovilay) [inline imports/plugins/custom/reviews/client/components/ReviewsPagination.js:12]: Great work so far. 
Very readable code. keep it up.

Just one question.
What if `totalProductReviews` is just one, do you still need to pluralize?, I can't see any check for the length of the `totalProductReviews`. 
on line 12.

**Comment by iverenshaguy** [comment]
Approved!

**Review by iverenshaguy — APPROVED** [review]

**Review by rovilay — APPROVED** [review]
Great work! keep it up.

**Pull request merged**
